Change order of parameters in invariant_violated_string constructor

NlightNFotis opened the pull request:
This is a trivial change, fixing a small regression introduced during a previous refactoring. Without it you get the wrong condition and reason when an invariant is getting triggered, like this:

```
Condition: true case of to_if_expr should be of same type as false case
Reason: expr.op1().type() == expr.op2().type()
```

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

NlightNFotis committed:
Change order of parameters in invariant_violated_string constructor

This is a trivial change, fixing a small regression introduced during a
previous refactoring. Without it you get the wrong condition and reason
when an invariant is getting triggered, like this:

Condition: true case of to_if_expr should be of same type as false case
Reason: expr.op1().type() == expr.op2().type()

tautschnig commented:
I have re-added the PR template, with two tasks still open.
1) The PR description has a very informative message, please make sure this makes it into the commit message.
2) This PR shows how important tests are... - please add one.

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 69b42d01ba9f402626509d98c009c56646abf559).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85582980

NlightNFotis committed:
Modify first invariant test to check for correct ordering of condition and reason of failure

NlightNFotis commented:
> This PR shows how important tests are... - please add one.

Agreed. Thankfully there were already tests, it was just that they weren't testing for this. I have updated them. I have also updated the commit message of the first commit to also contain the message in the PR. Can you take another look now @tautschnig ?

hannes-steffenhagen-diffblue commented on regression/invariants/invariant-failure/test.desc:
```diff
@@ -7,6 +7,8 @@ string
 Invariant check failed
 ^(Backtrace)|(Backtraces not supported)$
+^Condition: false$
+^Reason: Test invariant failure$
```
I'd strongly prefer adding another test for this rather than piggy-backing off an existing one (basically this tests two different things now)
(Not a blocker, just a preference)

NlightNFotis replied:
I am fairly certain this is the place for this to be added. The first test doesn't seem to do anything fairly important otherwise, and this is just adding some extra lines that should be there and are standard for every invariant failure.

danpoe reviewed (approved):
LGTM

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: 1ba7d157beb933d7c666afef59e52570edbbd88c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85752959
